lib/languages/js/source-code/source-code.js

```diff
@@ -476,18 +477,39 @@ class SourceCode extends TokenStore {
             ["configNodes", void 0]
         ]);
 
+        /**
+         * Indicates if the AST is ESTree compatible.
+         * @type {boolean}
+         */
+        this.isESTree = ast.type === "Program";
+
+
+        /*
+         * Backwards compatibility for BOM handling.
+         *
+         * The `hasBOM` property has been available on the `SourceCode` constructor
+         * for a long time and is used to indicate if the source contains a BOM.
+         * The linter strips the BOM and just passes the `hasBOM` property to the
+         * `SourceCode` constructor to make it easier for languages to not deal with
+         * the BOM.
```

Comment:
I got confused by this explanation.

1. This states that the `hasBOM` property has been available on the `SourceCode` constructor for a long time, but we're adding it now so it hasn't been available for a long time.
2. Linter passes `hasBOM` (on `file`) to `language.createSourceCode()`, not directly to the `SourceCode` constructor.

Reply:
Got it, but "property has been available on the `SourceCode` constructor`" sounded to me more like a property in the options object that is passed to the constructor rather than a property that the constructor creates on instances.

Reply:
Yeah, "constructor" should be "object".